### ISSUE TITLE
Signed Windows Installers

### ISSUE THREAD

**Issue by tresf** [qzind/qz-print#80]
Placholder for signing our Windows installers to prevent warning dialogs.

![image](https://cloud.githubusercontent.com/assets/6345473/10541666/e0f0bff6-73e0-11e5-9e47-9f9022e07923.png)

According to NSIS's documentation, this seems possible via [`signtool.exe`](http://nsis.sourceforge.net/Run_Command_After_Compilation) on Windows and [`signcode`](http://manpages.ubuntu.com/manpages/dapper/man1/signcode.1.html) on Linux.

**Linux Signing**

``` bash
# Do key conversion
openssl pkcs12 -in key-pair.pfx -nocerts -nodes -out private-key.pem
openssl rsa -in private-key.pem -outform PVK -pvk-strong -out private-key.pvk
openssl pkcs12 -in key-pair.pfx -nokeys -nodes -out certificate.pem
openssl crl2pkcs7 -nocrl -certfile certificate.pem -outform DER -out certificate.spc

# Linux Signing - Provided with mono-dev (sudo apt-get install mono-devel)
signcode -spc certificate.spc -v private-key.pvk -a sha1 -$ commercial -n "${socket.name}" -i \
${vendor.website} -t ${signing.tsaurl} -tr 10 "${out.dir}\${build.socket.name}-${build.version}.exe"

# Clean up files
rm private-key.pvk certificate.pem certificate.spc
```

**Windows Signing**

``` bash
# Provided with Windows SDK (https://msdn.microsoft.com/en-us/windows/desktop/bg162891.aspx)
signtool.exe sign /f key-pair.pfx /p pass /t ${signing.tsaurl} "${out.dir}\${build.socket.name}-${build.version}.exe
```

**Comment by tresf** [comment]
FYI, @Vzor- has started this.

**Edit:** Added properties example.

This is placed in `~/Desktop/Code Signing/private.properties` and is automatically loaded via [`build.xml#L17`](https://github.com/qzind/qz-print/blob/1.9/build.xml#L17)

``` properties
signing.alias=my-java-keystore-alias
signing.tsaurl=http://tsa.starfieldtech.com
signing.keystore=C:\\Full\\Path\\To\\Keystore.ks
signing.keypass=somekeypass
signing.storepass=somekeypass
```

perhaps add

``` properties
signing.pfx=C:\\Full\\Path\\To\\Keypair.pfx
signing.pfxpass=somepfxpass
```

**Comment by tresf** [comment]
Closed via 79ca3b2.  @Vzor- did this in a way which doesn't require any special pfx certs. :+1:

**Comment by ordercup** [comment]
Great! :-)

Aloke Nath | OrderCup
www.ordercup.com
D: 949-482-6186
O: 877-240-3545
